build/gulpfile.vscode.linux.js

```diff
@@ -142,6 +175,76 @@ function buildRpmPackage(arch) {
 	]);
 }
 
+function getFlatpakArch(arch) {
+	return { x64: 'x86_64', ia32: 'i386', arm: 'arm' }[arch];
+}
+
+function prepareFlatpak(arch) {
+	// This is not imported in the global scope to avoid requiring ImageMagick
+	// (or GraphicsMagick) when building not building Flatpak bundles.
+	const imgResize = require('gulp-image-resize');
+
+	const binaryDir = '../VSCode-linux-' + arch;
+	const flatpakArch = getFlatpakArch(arch);
+	const destination = '.build/linux/flatpak/' + flatpakArch;
+
+	return function () {
+		const all = [16, 24, 32, 48, 64, 128, 192, 256, 512].map(function (size) {
+			return gulp.src('resources/linux/code.png', { base: '.' })
+				.pipe(imgResize({ width: size, height: size, format: "png", noProfile: true }))
+				.pipe(rename('share/icons/hicolor/' + size + 'x' + size + '/apps/' + flatpakManifest.appId + '.png'));
+		});
+
+		all.push(gulp.src('resources/linux/code.desktop', { base: '.' })
+			.pipe(replace('Exec=/usr/share/@@NAME@@/@@NAME@@', 'Exec=code-oss'))
+			.pipe(replace('@@NAME_LONG@@', product.nameLong))
+			.pipe(replace('@@NAME_SHORT@@', product.nameShort))
+			.pipe(replace('@@NAME@@', product.applicationName))
+			.pipe(rename('share/applications/' + flatpakManifest.appId + '.desktop')));
+
+		all.push(gulp.src('resources/linux/code.appdata.xml', { base: '.' })
+			.pipe(replace('@@NAME_LONG@@', product.nameLong))
+			.pipe(replace('@@NAME@@', flatpakManifest.appId))
+			.pipe(rename('share/appdata/' + flatpakManifest.appId + '.appdata.xml')));
+
+		all.push(gulp.src(binaryDir + '/**/*', { base:binaryDir })
+			.pipe(rename(function (p) {
+				p.dirname = 'share/' + product.applicationName + '/' + p.dirname;
+			})));
+
+		return es.merge(all).pipe(vfs.dest(destination));
+	}
+}
+
+function buildFlatpak(arch) {
+	const flatpakArch = getFlatpakArch(arch);
+	const manifest = {};
+	for (var k in flatpakManifest) manifest[k] = flatpakManifest[k];
```

Comment:
Use braces and put `manifest[k]..` on new line as this would fail ts style rules